lib/atscppapi/src/Transaction.cc

```diff
@@ -186,6 +186,20 @@ Transaction::setErrorBody(const std::string &page)
   TSHttpTxnErrorBodySet(state_->txn_, TSstrdup(page.c_str()), page.length(), NULL); // Default to text/html
 }
 
+void
+Transaction::setErrorBody(const std::string &page, const std::string &mimetype)
+{
+  LOG_DEBUG("Transaction tshttptxn=%p setting error body page: %s", state_->txn_, page.c_str());
+  TSHttpTxnErrorBodySet(state_->txn_, TSstrdup(page.c_str()), page.length(), TSstrdup(mimetype.c_str()));
```

Comment:
I'm totally good w/ this addition but there is a bug that was recently identified in `Transaction::setErrorBody()`, if someone passes a binary string in for page it will result in weird truncation because of the use of string functions. The code needs to change to be:

`
char *body = (char*)TSmalloc(page.length());
memcpy(body, page.data(), page.length());
TSHttpTxnErrorBodySet(state_->txn_, body, page.length(), TSstrdup(mimetype.c_str()));
`

Can you make this change to both versions of `Transaction::setErrorBody()`?